### PR TITLE
fixes #9812 - Adds validation on override value, ensure match returns er...

### PR DESCRIPTION
**Pull request by shlomizadok**
...ror if blank

### CODE REVIEW

**Comment by shlomizadok** [comment]
[test]

File: test/functional/api/v2/override_values_controller_test.rb [hunk 1]
```diff
@@ -67,4 +67,26 @@ class Api::V2::OverrideValuesControllerTest < ActionController::TestCase
...
+    assert_difference('LookupValue.count', 0) do
+      post :create,  {:smart_variable_id => k.id, :override_value => { :value => 'xyz=10' } }
+    end
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert_equal("Validation failed: Match can't be blank", response['error']['message'])
+    assert_response :error
+  end
+
+  test "should not create override value without match" do
+    k = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
```
Comment (dLobatog) [inline test/functional/api/v2/override_values_controller_test.rb]: Almost the same preparation code as the other test, can you DRY it a bit? It looks like you can do something like

``` ruby
[{ :value => 'xyz=10'}, { :match => 'string'}].each do |override_value|
  test "should not create override value without #{override_value.key}" do
    lookup_key = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
    refute lookup_key.override
    assert_difference('LookupValue.count', 0) do
      post :create, { :smart_variable_id => lookup_key.id, :override_value => override_value }
    end
    response = ActiveSupport::JSON.decode(@response.body)
    assert_matches /#{override_value.key} can't be blank/, response['error']['message']
    assert_response :error
  end
end
```

File: test/functional/api/v2/override_values_controller_test.rb [hunk 2]
```diff
@@ -67,4 +67,26 @@ class Api::V2::OverrideValuesControllerTest < ActionController::TestCase
     end
     assert_response :success
   end
+
+  test "should not create override value without value" do
+    k = FactoryGirl.create(:lookup_key, :puppetclass => puppetclasses(:two))
```
Comment (dLobatog) [inline test/functional/api/v2/override_values_controller_test.rb]: Isn't `lookup_key` more readable for newcomers?

**Comment by dLobatog** [comment]
@shlomizadok Thanks for the contribution, it's ready to merge after fixing the test in my opinion. :shipit:  

If you want to go the extra mile, `ensure_xxxx_exists` needs refactoring as well... :eyes:

**Comment by shlomizadok** [comment]
@eLobato thanks for reviewing. I did some refactoring on `ensure_xxxx_exists` let me know what you think.

**Comment by shlomizadok** [comment]
[test] cause robucop was [aborted](http://ci.theforeman.org/job/test_develop_pull_request/10750/test=rubocop/console)

**Comment by dLobatog** [comment]
~~@shlomizadok Can you separate the refactor to other PR? The doc change is mergeable right away + the issue number for this doesn't have to do with the refactor. I appreciate it but let's keep PRs separate~~ Nevermind I thought this PR was https://github.com/theforeman/foreman/pull/2263#issuecomment-85508107

**Comment by domcleal** [comment]
Could you rebase + squash please?  The same method was changed in another recent PR.

**Comment by shlomizadok** [comment]
rebased and squashed (thanks for the heads up on  method change)

**Comment by dLobatog** [comment]
Merged as eaab1bb81ade14719059006947495a2073b6e0f5, thanks @shlomizadok!

**Pull request closed without merging**